src/EditorFeatures/CSharpTest/ImplementAbstractClass/ImplementAbstractClassTests.cs

```diff
@@ -1818,6 +1818,172 @@ public override void M<T>()
     {
         throw new System.NotImplementedException();
     }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementAbstractClass)]
+        public async Task TestFlowAnalysisAttributeOnNonNullableReferenceType()
+        {
+            await TestInRegularAndScript1Async(
+@"#nullable enable
+
+abstract class Base<T>
+{
+    [return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
+    protected abstract T MethodAllowNull([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+
+    [return: System.Diagnostics.CodeAnalysis.NotNullAttribute]
+    protected abstract T MethodDisallowNull([System.Diagnostics.CodeAnalysis.DisallowNullAttribute] T value);
+}
+
+class [|Class|] : Base<string>
+{
+}",
+@"#nullable enable
+
+abstract class Base<T>
+{
+    [return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
+    protected abstract T MethodAllowNull([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+
+    [return: System.Diagnostics.CodeAnalysis.NotNullAttribute]
+    protected abstract T MethodDisallowNull([System.Diagnostics.CodeAnalysis.DisallowNullAttribute] T value);
+}
+
+class Class : Base<string>
+{
+    protected override string MethodAllowNull(string? value)
```

Comment:
❕ The return value here is not correct.

```suggestion
    protected override string? MethodAllowNull(string? value)
```